Improvements to `deploy` and `preview` commands

vernig opened the pull request:
When using Pulumi with local service, a passphrase needs to be generated and used for each command.
Since the pulumi CLI is executed in non interactive mode, we need to pass the passphrase as env variable.

As first step, the code is checking if Pulumi is running with local service (using `pulumi login` output). If it does, it prompts the user to provide the passphrase and add that to the env variables

Fix #1
Fix #2

Commit:
Fix: Ask for passphrase if pulumi uses local

When using Pulumi with local service, a passphrase needs
to be generated and used for each command.
Since the pulumi CLI is executed in non interactive mode,
we need to pass the passphrase as env variable.
As first step, the code is checking if Pulumi is running with local
(using `pulumi login` output). If it does, it prompts the user
to provide the passphrase and add that to the env variables

Fix #1
Fix #2

vernig commented:
@kaiquelupo can you merge the PRs from https://github.com/kaiquelupo/plugin-twilio-infra so you can test this changes as well?

vernig committed:
Merge pull request #8 from twilio-infra-as-code/main

Merge upstream changes

Commit:
Fix missing curly braces at the end of setStack

bermudezmt commented on src/utils.js:
```diff
@@ -130,20 +138,58 @@ function setStack(flags, args, clientCommand) {
+    const answers = await inquirer.prompt([
+      {
+        type: "password",
+        name: "passPhrase",
```
Could this in theory also work if we changed the name to `passphrase` and `answers.passphrase`? I was just wondering why it's using camelcase.

vernig replied:
Good point @bermudezmt . Done in e1a6f3e

bermudezmt reviewed (approved):
I can confirm that it works, but we should probably actually set the passphrase as an environment variable after entering it on stack initialization. Right now, every invocation of a twilio infra subcommand requires a passphrase when using local storage.

Thanks for your work, @vernig !

Commit:
Check if passphrase is already set as env variable

If it's set, then don't ask the user to enter it

Commit:
Fix runPulumiCommand is an async function

Commit:
Check stack flag for preview and deploy command

Commit:
Add a logger function

TBD: to improve for warning, logs, info etc...

Commit:
Improvement to checkPulumi and error handling

Commit:
Improve erro handling for runPulumiCommand

Commit:
Remove camelcase from passphrase

vernig commented:
> I can confirm that it works, but we should probably actually set the passphrase as an environment variable after entering it on stack initialization. Right now, every invocation of a twilio infra subcommand requires a passphrase when using local storage.

This is a very good point @bermudezmt. I fixed it in https://github.com/twilio-infra-as-code/plugin-twilio-infra/pull/3/commits/dbeafaa04424c2e251c1ad933c0d0c4b148209e9 but didn't thoroughly tested yet. Let me know if you get a chance to do it. 

Also, I added some more proper error handling and made the `--stack` flag required for `preview` and `deploy`
